### ISSUE TITLE
Improve CSP on the public testpilot site

### ISSUE THREAD

**Issue by jvehent** [mozilla/testpilot#1311]
Testpilot gets an F on the observatory testing: https://observatory.mozilla.org/analyze.html?host=testpilot.firefox.com The CSP is particularly bad and should be improved.
1. Remove all the HTTP includes (everything must use HTTPS).
2. Limit the mozilla domains allowed to insert scripts and styles, and not just use all `*.mozilla.{net,org}`.
3. Don't allow script-src unsafe-inline. This just makes CSP useless.
4. Remove -dev domains on the production site. dev domains have lower security than production sites and should not be allowed to inject content in prod.
5. add a report uri to `/__cspreport__`
6. set `frame-src` to self

Current policy shown below:

```
Content-Security-Policy: 
    img-src 'self'
        http://*.mozilla.net
        https://*.mozilla.net
        http://*.mozilla.org
        https://*.mozilla.org
        http://*.gravatar.com
        https://*.gravatar.com
        http://*.google-analytics.com
        https://*.google-analytics.com
        https://ssl.gstatic.com/
        https://testpilot-dev-media.s3.amazonaws.com
        https://testpilot-stage.s3.amazonaws.com
        https://testpilot-prod.s3.amazonaws.com;
    connect-src 'self'
        http://*.mozilla.net
        https://*.mozilla.net
        http://*.mozilla.org
        https://*.mozilla.org;
    script-src 'self'
        http://*.mozilla.org
        https://*.mozilla.org
        http://*.mozilla.net
        https://*.mozilla.net
        http://*.google-analytics.com
        https://*.google-analytics.com
        https://apis.google.com;
    default-src 'self';
    font-src 'self'
        http://*.mozilla.net
        https://*.mozilla.net
        http://*.mozilla.org
        https://*.mozilla.org;
    style-src 'self'
        'unsafe-inline'
        http://*.mozilla.org
        https://*.mozilla.org
        http://*.mozilla.net
        https://*.mozilla.net
```

Proposed policy:

```
Content-Security-Policy: 
    img-src 'self'
        https://*.mozilla.net
        https://*.mozilla.org
        https://*.gravatar.com
        https://ssl.gstatic.com/
        https://testpilot-prod.s3.amazonaws.com;
    connect-src 'self';
    script-src 'self'
        https://*.mozilla.net
        https://*.google-analytics.com
        https://apis.google.com;
    default-src 'self';
    font-src 'self'
        https://*.mozilla.net
        https://*.mozilla.org;
    style-src 'self'
        https://*.mozilla.net;
    frame-src 'self';
    report-uri /__cspreport__;
```

cc @marumari

**Comment by lmorchard** [comment]
FWIW, CSP settings are here: https://github.com/mozilla/testpilot/blob/master/testpilot/settings.py#L327

I think we'll need to leave the HTTP CSP policy in, but only in the DEBUG=True case - local development & testpilot.dev.mozaws.net are both still HTTP (non-SSL) (see also #548)

**Comment by ghost** [comment]
This is blocked on us figuring out where we're going to put the headers.  S3/Cloudfront don't allow arbitrary headers.  Discussion started at https://mail.mozilla.org/pipermail/testpilot-dev/2016-September/000250.html

Since we'd still need to get all that infrastructure up before we can do this, we're going to be squished for getting this done in this milestone.  

/cc @relud for great ideas :)

**Comment by ghost** [comment]
Alright, this is fixed in #1481 with a much tighter policy

And is being deployed in https://bugzilla.mozilla.org/show_bug.cgi?id=1306680
